WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt

```diff
@@ -496,36 +521,58 @@ class PostListViewModel @Inject constructor(
         _postListAction.postValue(PostListAction.EditPost(site, post))
     }
 
-    private fun trashPost(localPostId: Int) {
+    private fun deletePost(localPostId: Int) {
         // If post doesn't exist, nothing else to do
         val post = postStore.getPostByLocalPostId(localPostId) ?: return
-        postIdToTrash = Pair(LocalPostId(LocalId(post.id)), RemotePostId(RemoteId(post.remotePostId)))
-        // Refresh the list so we can immediately hide the post
-        pagedListWrapper.invalidateData()
-        val undoAction = {
-            postIdToTrash = null
-            // Refresh the list, so we can re-show the post
-            pagedListWrapper.invalidateData()
-        }
-        val onDismissAction = {
-            // If postIdToTrash is set to `null`, user undid the action
-            if (postIdToTrash != null) {
-                postIdToTrash = null
-                _postUploadAction.postValue(CancelPostAndMediaUpload(post))
-                if (post.isLocalDraft) {
-                    val pushId = PendingDraftsNotificationsUtils.makePendingDraftNotificationId(post.id)
-                    _postListAction.postValue(DismissPendingNotification(pushId))
-                    dispatcher.dispatch(PostActionBuilder.newRemovePostAction(post))
-                } else {
-                    dispatcher.dispatch(PostActionBuilder.newDeletePostAction(RemotePostPayload(post, site)))
-                }
+
+        when {
+            post.isLocalDraft -> {
+                val pushId = PendingDraftsNotificationsUtils.makePendingDraftNotificationId(post.id)
+                _postListAction.postValue(DismissPendingNotification(pushId))
+                dispatcher.dispatch(PostActionBuilder.newRemovePostAction(post))
+            }
+            else -> {
+                dispatcher.dispatch(PostActionBuilder.newDeletePostAction(RemotePostPayload(post, site)))
```

Comment:
This action can take a while and the UI seems a bit unresponsive on slow connection. I think we should consider using the same "UX hack" as for "postsBeingTrashedOrRestored" and display Progress+Overlay. Wdyt?

![delete-post](https://user-images.githubusercontent.com/2261188/55866922-ac3b7f80-5b81-11e9-9bc0-b1502aa4436f.gif)

Reply:
Delete action is added to critical post actions in 9ee07273bd8d019fec5d29a8f0948defa1e0ec0c.